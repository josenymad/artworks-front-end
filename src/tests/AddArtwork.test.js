import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AddArtWork from "../components/AddArtwork";

describe("Add Artwork", () => {
  const formData = {
    company: "",
    product: "",
    partNumber: "",
    date: "",
    image: "",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AddArtWork formData={formData} />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders label for Company", () => {
    render(
      <BrowserRouter>
        <AddArtWork formData={formData} />
      </BrowserRouter>,
    );
    const companyLabel = screen.getByText("Company:");

    expect(companyLabel).toBeInTheDocument();
  });

  it("renders label for Product", () => {
    render(
      <BrowserRouter>
        <AddArtWork formData={formData} />
      </BrowserRouter>,
    );
    const productLabel = screen.getByText("Product:");

    expect(productLabel).toBeInTheDocument();
  });

  it("renders label for Part Number", () => {
    render(
      <BrowserRouter>
        <AddArtWork formData={formData} />
      </BrowserRouter>,
    );
    const partNumberLabel = screen.getByText("Part Number:");

    expect(partNumberLabel).toBeInTheDocument();
  });

  it("renders label for Date", () => {
    render(
      <BrowserRouter>
        <AddArtWork formData={formData} />
      </BrowserRouter>,
    );
    const dateLabel = screen.getByText("Date:");

    expect(dateLabel).toBeInTheDocument();
  });

  it("renders label for Image", () => {
    render(
      <BrowserRouter>
        <AddArtWork formData={formData} />
      </BrowserRouter>,
    );
    const imageLabel = screen.getByText("Image:");

    expect(imageLabel).toBeInTheDocument();
  });

  it("renders CLEAR button", () => {
    render(
      <BrowserRouter>
        <AddArtWork formData={formData} />
      </BrowserRouter>,
    );
    const clearButton = screen.getByText("CLEAR");

    expect(clearButton).toBeInTheDocument();
  });
});
