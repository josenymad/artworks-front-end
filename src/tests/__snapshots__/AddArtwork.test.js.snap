// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Artwork renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-artwork"
  >
    <form
      class="add-artwork__form"
      enctype="multipart/form-data"
    >
      <label
        class="add-artwork__form__label"
        for="company"
      >
        Company:
      </label>
      <input
        class="add-artwork__form__input"
        id="company"
        maxlength="36"
        name="company"
        placeholder="Certags"
        required=""
        value=""
      />
      <label
        class="add-artwork__form__label"
        for="product"
      >
        Product:
      </label>
      <input
        class="add-artwork__form__input"
        id="product"
        maxlength="36"
        name="product"
        placeholder="size, material, detail"
        required=""
        value=""
      />
      <label
        class="add-artwork__form__label"
        for="part-number"
      >
        Part Number:
      </label>
      <input
        class="add-artwork__form__input"
        id="part-number"
        maxlength="36"
        name="partNumber"
        placeholder="CERUK0123"
        value=""
      />
      <label
        class="add-artwork__form__label"
        for="date"
      >
        Date:
      </label>
      <input
        class="add-artwork__form__input"
        id="date"
        name="date"
        required=""
        type="date"
        value=""
      />
      <label
        class="add-artwork__form__label"
        for="image"
      >
        Image:
      </label>
      <input
        accept=".pdf"
        class="add-artwork__form__input"
        id="image"
        name="image"
        required=""
        type="file"
      />
      <div
        class="add-artwork__form-buttons"
      >
        <button
          class="add-artwork__form__button"
          type="button"
        >
          CLEAR
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
